src/commands/createNewProject/javaSteps/JavaVersionStep.ts

```diff
@@ -4,29 +4,46 @@
  *--------------------------------------------------------------------------------------------*/
 
 import { AzureWizardPromptStep, IAzureQuickPickItem } from "@microsoft/vscode-azext-utils";
+import { previewDescription } from "../../../constants";
 import { hasMinFuncCliVersion } from "../../../funcCoreTools/hasMinFuncCliVersion";
 import { localize } from "../../../localize";
 import { IJavaProjectWizardContext } from "./IJavaProjectWizardContext";
 
 export const java8: string = '8';
 export const java11: string = '11';
+export const java17: string = '17';
+
+const versionInfo: [string, string, string, string?][] = [
```

Comment:
I think it'd be easier to read if you made a type here. It can make your `getPicks` cleaner too because you could just pass the object type into the results array.

For example, if you make the type like this:
```
type javaVersionInfo = {
label: string,
data: string,
description?: string,
miniFunc: string
}
```

It could be pushed into the array and be used as a quickPickItem directly.

Reply:
Thanks for your advices, related code has been updated